// octavian/build/chord/modes.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var modes = Object.defineProperties({
  major: ['majorThird', 'perfectFifth'],
  majorSixth: ['majorThird', 'perfectFifth', 'majorSixth'],
  majorSeventh: ['majorThird', 'perfectFifth', 'majorSeventh'],
  majorSeventhFlatFive: ['majorThird', 'diminishedFifth', 'majorSeventh'],
  majorSeventhSharpFive: ['majorThird', 'minorSixth', 'majorSeventh'],
  minor: ['minorThird', 'perfectFifth'],
  minorSixth: ['minorThird', 'perfectFifth', 'majorSixth'],
  minorSeventh: ['minorThird', 'perfectFifth', 'minorSeventh'],
  minorMajor: ['minorThird', 'perfectFifth', 'minorSeventh'],
  dominantSeventh: ['majorThird', 'perfectFifth', 'minorSeventh'],
  diminished: ['minorThird', 'diminishedFifth'],
  diminishedSeventh: ['minorThird', 'diminishedFifth', 'majorSixth'],
  halfDimished: ['minorThird', 'diminishedFifth', 'minorSeventh'] }, {
  maj: {
    get: function get() {
      return this.major;
    },
    enumerable: true
  },
  '6': {
    get: function get() {
      return this.majorSixth;
    },
    enumerable: true
  },
  maj6: {
    get: function get() {
      return this.majorSixth;
    },
    enumerable: true
  },
  '7': {
    get: function get() {
      return this.majorSeventh;
    },
    enumerable: true
  },
  maj7: {
    get: function get() {
      return this.majorSeventh;
    },
    enumerable: true
  },
  maj7b5: {
    get: function get() {
      return this.majorSeventhFlatFive;
    },
    enumerable: true
  },
  'maj7#5': {
    get: function get() {
      return this.majorSeventhSharpFive;
    },
    enumerable: true
  },
  min: {
    get: function get() {
      return this.minor;
    },
    enumerable: true
  },
  m: {
    get: function get() {
      return this.minor;
    },
    enumerable: true
  },
  min6: {
    get: function get() {
      return this.minorSixth;
    },
    enumerable: true
  },
  m6: {
    get: function get() {
      return this.minorSixth;
    },
    enumerable: true
  },
  min7: {
    get: function get() {
      return this.minorSeventh;
    },
    enumerable: true
  },
  m7: {
    get: function get() {
      return this.minorSeventh;
    },
    enumerable: true
  },
  'm#7': {
    get: function get() {
      return this.minorMajor;
    },
    enumerable: true
  },
  'min#7': {
    get: function get() {
      return this.minorMajor;
    },
    enumerable: true
  },
  'm(maj7)': {
    get: function get() {
      return this.minorMajor;
    },
    enumerable: true
  },
  dom7: {
    get: function get() {
      return this.dominantSeventh;
    },
    enumerable: true
  },
  dim: {
    get: function get() {
      return this.diminished;
    },
    enumerable: true
  },
  dim7: {
    get: function get() {
      return this.diminishedSeventh;
    },
    enumerable: true
  },
  m7b5: {
    get: function get() {
      return this.halfDiminshed;
    },
    enumerable: true
  }
});

exports['default'] = Object.freeze(modes);
module.exports = exports['default'];
